tests/dnfserver/container/Dockerfile

```diff
@@ -0,0 +1,9 @@
+
+FROM registry.fedoraproject.org/fedora:40
+RUN dnf -y install openssl nginx nano procps-ng python3-pip
+COPY certificates /certificates
+COPY setup.sh .
+RUN ./setup.sh
+COPY nginx.conf /etc/nginx/conf.d/
+EXPOSE 8443 8080 8081
+ENTRYPOINT /usr/sbin/nginx -g 'daemon off;'
```

Comment:
I'm not convinced we need this at all.
Instead, I think you should use the upstream `nginx:alpine-slim` image (~12MB) and just mount a bunch of files into it read-only.
You're proposing an nginx config and certificates, so the only bit missing is creating a DNF repo out of some RPM files. I don't think we want to build a new image based on Fedora (not even minimal) just to do this based on the fact that `createrepo_c` is already cachi2's direct dependency, so it'll be always present in the test environment. So I imagine the RPM will be downloaded in the `start.sh` script in the `setup()` phase just like the pypiserver does it and then all the cert files, config and the dnf repo would be mounted to the contianer.

Reply:
I didn't choose Fedora because of createrepo_c (although it is convenient). I don't recall the exact detail but it was about needing the newer version of nginx which is in Fedora. I think it could be moved to ubi10 when that is out and get same version.

Personally I still think being able to run rpm_lockfile_prototype or whatever replaces it against the container itself is ideal.  If you worry about things like the size of container images (I don't very much) then I guess we could use alpine. I prefer Fedora because I more or less know how it works, I know who can fix it when it doesn't work and it is easy to install other packages or tools when the need arises.

Example: in 2025 we probably want to test this with quantum safe certificates - I know openssl and nginx will be updated appropriately here because I am on the project. I don't know anything about Alpine and I'm not eager to touch something again which is working.

Reply:
>  I don't recall the exact detail but it was about needing the newer version of nginx which is in Fedora

Newer than the official nginx docker image (1.26/1.27)?

> Personally I still think being able to run rpm_lockfile_prototype or whatever replaces it against the container itself is ideal

How is ^this relevant to the integration test suite?

> I prefer Fedora because I more or less know how it works, I know who can fix it when it doesn't work and it is easy to install other packages or tools when the need arises.
...
I don't know anything about Alpine and I'm not eager to touch something again which is working.

The thing about application containers is that you shouldn't need to, what platform nginx is running is irrelevant to the cause, because as a consumer of the image you only care about the application itself unless you need to install extra packages (which is where you might start caring) which you likely don't need here. As long as you have an nginx config and a volume for data, I don't see how quantum safe certificates are an issue here, because nginx doesn't care as long as it can find them in some filesystem location it can read. Therefore I'm not convinced any of what you mentioned is a real problem that would justify our integration test suite to build a dedicated container image for its own use instead of pulling official image builds suited for the workload.

The downside of stuffing all the logic revolving around `createrepo_c` to the container build time as you currently have is that you have limited visibility into it and capabilities to debug the process of creation of the data during the build. Conversely, if you mount all of the data merely as a data volume to an nginx container, you can both issue breakpoints() to the relevant Python code that generates/handles them or inspect it from the host easily (i.e. you don't need to spawn another interactive shell session in the container).